### PR TITLE
New build system

### PR DESCRIPTION
**Pull request by SferaDev**
- Unify package manager in all repos
- Speed up dependency gathering in CI
- Fix builds and failed deployments in CI
- Make build step faster with `turbo`

### CODE REVIEW

**Comment by SferaDev** [comment]
This is ready to review, `size-limit` GH action will fail until we merge it. It's failing now when going to main and not resolving some dependencies with `pnpm`.

**Review by richardgill — APPROVED** [review]
This look good. Agreed to merge at 11am Berlin today

File: .github/workflows/build-pr.yml [hunk 1]
```diff
@@ -84,7 +90,23 @@ jobs:
...
+          '@xata.io/codegen': patch
+          '@xata.io/client': patch
+          '@xata.io/cli': patch
+          ---
+          Force canary build
+
+          EOF
+
+          npx changeset version --snapshot ${{ github.ref_name == 'changeset-release/main' && 'beta' || 'alpha' }}
+          npx changeset publish --tag ${{ github.ref_name == 'changeset-release/main' && 'dev' || 'canary' }} --no-git-tag
```
Comment (SferaDev) [inline .github/workflows/build-pr.yml:109]:
```suggestion
          pnpm dlx changeset version --snapshot ${{ github.ref_name == 'changeset-release/main' && 'beta' || 'alpha' }}
          pnpm dlx changeset publish --tag ${{ github.ref_name == 'changeset-release/main' && 'dev' || 'canary' }} --no-git-tag
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,20 +29,27 @@ jobs:
...
+        run: pnpm install --frozen-lockfile
+
+      - name: Build
+        run: pnpm build
 
       - name: Create Release Pull Request or Publish to npm
         uses: changesets/action@v1
         with:
           title: Release tracking
-          publish: npx changeset publish
```
[the remark this answers is not in the document]
Comment (SferaDev) [reply .github/workflows/release.yml:45]: No, changesets does the publishing step. We had a customized script here but wasn't needed. Removed for clean-up

File: turbo.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "tsc": {
+      "dependsOn": ["add-version", "^tsc"],
+      "inputs": ["src/**"],
+      "outputs": []
+    },
+    "build": {
+      "dependsOn": ["add-version", "^build"],
+      "inputs": ["src/**"],
+      "outputs": ["dist/**"]
+    }
```
[the remark this answers is not in the document]
Comment (SferaDev) [reply turbo.json:16]: They're not part of the build system nor required to be checked for failures.

For size limit we check it on each PR with a separate workflow.

For E2E tests we only check them once per release (they are very expensive), so we don't check every time just before releasing something to the world

**Comment by SferaDev** [comment]
@atilafassina Since we can continue using `npx` without problems I'd keep with it for now and change them afterwards one by one.

**Pull request merged**